test/async-hooks/test-zctx.js

```diff
@@ -0,0 +1,31 @@
+const common = require('../common')
+const assert = require('assert')
+const initHooks = require('./init-hooks')
+const { checkInvocations } = require('./hook-checks')
+const zlib = require('zlib')
+
+const hooks = initHooks()
+
+hooks.enable()
+zlib.deflate('..........', common.mustCall(ondeflate));
+
+function ondeflate() {
+  const as = hooks.activities()
+  const a = as[0]
+  checkInvocations(a, { init: 1, before: 2, after: 2 }, 'while in ondeflate callback')
+}
+
+process.on('exit', onexit)
+
+function onexit() {
+  hooks.disable()
+  hooks.sanityCheck()
+
+  const as = hooks.activities()
+  assert.strictEqual(as.length, 6, 'six activities')
```

Comment:
OK @Fishrock123 wasn't aware, so obviously then `strictEqual` is fine, even if it's total overkill in those cases.
Not sure why the core team made the decision to use `strictEqual` everywhere (even for primitives).